### PR TITLE
Guile deps

### PR DESCRIPTION
**Pull request by stikonas**
These are dependencies needed for Guile 3.0.7.

I have a Guile build (with psyntax-pp boostrap) too but at the moment 2 files are not reproducible.

### CODE REVIEW

File: sysa/gc-8.0.4/gc-8.0.4.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+src_configure() {
+    # CFLAGS needed on musl
+    ./configure \
+	--prefix="${PREFIX}" \
+	--build=i386-unknown-linux-gnu \
+	--host=i386-unknown-linux-gnu \
+	--target=i386-unknown-linux-gnu \
+	--libdir="${PREFIX}/lib/musl" \
+	--disable-shared \
+	CFLAGS='-D_GNU_SOURCE -DNO_GETCONTEXT -DSEARCH_FOR_DATA_START -DUSE_MMAP -DHAVE_DL_ITERATE_PHDR'
```
Comment (fosslinux) [inline sysa/gc-8.0.4/gc-8.0.4.sh:18]: 4 space indentation?

File: sysa/gcc-4.7.4/pass1.sh [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        cd ..
+    done
+    cd ..
+}
+
+src_compile() {
+    ln -s . build/build-i386-unknown-linux-gnu
+    for dir in libiberty libcpp libdecnumber gcc; do
+	# We have makeinfo now but it is not happy with gcc .info files, so skip it
+        make -C build/$dir LIBGCC2_INCLUDES=-I"${PREFIX}/include" STMP_FIXINC= GMPLIBS="-lmpc -lmpfr -lgmp" MAKEINFO=true
```
Comment (fosslinux) [inline sysa/gcc-4.7.4/pass1.sh:105]: Indentation, also line breaks?

File: sysa/libatomic_ops-7.6.10/libatomic_ops-7.6.10.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+src_configure() {
+    ./configure \
+	--prefix="${PREFIX}" \
+	--build=i386-unknown-linux-gnu \
+	--host=i386-unknown-linux-gnu \
+	--target=i386-unknown-linux-gnu \
+	--libdir="${PREFIX}/lib/musl" \
+	--disable-shared
```
Comment (fosslinux) [inline sysa/libatomic_ops-7.6.10/libatomic_ops-7.6.10.sh:16]: Indentation

**Review by fosslinux — CHANGES_REQUESTED** [review]
Still need to look over for pregened files but a quick look doesn't look too problematic.

**Comment by stikonas** [comment]
Fixed raised issues. Yeah, most of this software is non-GNU, so somehow have fewer pre-gen files. GCC is the biggest one here but it's more or less the same as GCC 4.0.4. There are some issues while rebuilding some configure files in GCC for now (probably need to fix aclocal parameters) but they are in the directories that we don't use for C only build. Once we do full GCC with C and C++ backends, we can fix those issues and backport it here.

**Comment by andriuss-xilinx** [comment]
I've resolved Guile reproducibility problems, so adding it too to this PR.

**Comment by stikonas** [comment]
I've resolved Guile reproducibility issues, so adding it to this PR too.

**Comment by stikonas** [comment]
Let's postpone the merge in the short term. We can merge this later. I'll cherry-pick download function fix though.

**Comment by stikonas** [comment]
Closing this for now. Some commits will be incorporated into musl_trippled PR, let's do rest of Guile after g++ (it will let us drop some patches too).

**Pull request closed without merging**